File: tests/fixtures/java/policies/pii_redaction_before_llm/compliant/com/example/app/Handler.java
package com.example.app;

import com.example.redactor.Redactor;
import com.example.anthropic.Anthropic;

public class Handler {
    private final Redactor redactor;
    private final Anthropic client;

    public Handler(Redactor redactor, Anthropic client) {
        this.redactor = redactor;
        this.client = client;
    }

    public String summarize(String userId) {
        var user = loadUser(userId);
        var safe = redactor.redact(user);
        return client.messagesCreate(safe);
    }

    private java.util.Map<String, String> loadUser(String userId) {
        return java.util.Map.of("id", userId, "email", "x@example.com");
    }
}

File: tests/fixtures/go/policies/pii_redaction_before_llm/compliant/handler/handler.go
//go:build policyguard_fixture

package handler

import (
	"redactor"

	llm "github.com/anthropic/anthropic-sdk-go"
)

func SummarizeUser(userID string) string {
	user := loadUser(userID)
	safe := redactor.Redact(user)
	return llm.Messages.Create(safe)
}

func loadUser(userID string) map[string]string {
	return map[string]string{"id": userID, "email": "x@example.com"}
}
